api/remoterelations/remoterelations_test.go

```diff
@@ -424,3 +425,54 @@ func (s *remoteRelationsSuite) TestSetRemoteApplicationStatus(c *gc.C) {
 	c.Check(err, gc.ErrorMatches, "FAIL")
 	c.Check(callCount, gc.Equals, 1)
 }
+
+type facadeCallFunc = func(objType string, version int, id, request string, arg, result interface{}) error
+
+func (s *remoteRelationsSuite) TestUpdateControllerForModelResultCount(c *gc.C) {
+	apiCaller := testing.APICallerFunc(
+		func(objType string, version int, id, request string, arg, result interface{}) error {
+			c.Assert(request, gc.Equals, "UpdateControllersForModels")
+			*(result.(*params.ErrorResults)) = params.ErrorResults{
```

Comment:
I'm not a fan of doing this, this pointer arithmetic is really painful to grok compared to mocks.

Reply:
Yeah, this is the old-style caller mocking. Suite is a little too large to re-write for one test. It is also in-flux with Christian's patches.